### ISSUE TITLE
Display email signature images

### ISSUE THREAD

**Issue by ManiMuridi** [jamesiarmes/php-ews#381]
Version (e.g. 1.0, dev-master): dev-master

Microsoft Exchange version: 2007,2013

Description of problem:
How can I display the image of an email signature I've received from a sender?
![signature](https://cloud.githubusercontent.com/assets/1406117/21798705/fe30423a-d726-11e6-8383-4733b6080c20.jpg)

**Comment by jamesiarmes** [comment]
I haven't verified, but I would expect there to be an attachment on the message for the inline image. You would then need to use a `GetAttachment` operation to download the file.

**Comment by ManiMuridi** [comment]
There is an attachment on the message and I did use the `GetAttachment` but when I download I have to change the url of the image in the message to display it properly on my email client. But does this mean that I have to resend that attachment using the original image src url that exchange created or can I just send it as an attachment again? I'll test it now but I wasn't sure if you worked in this section yet. I would love to contribute to this project as well so I will find some time soon for sure.

**Comment by jamesiarmes** [comment]
I'm not sure I understand your question. Are you asking if you need to re-attach the image in a reply that would include the original signature?

**Comment by ManiMuridi** [comment]
Basically, suppose I receive an email and there is a signature in the email which contains an image. The image doesn't display properly since its src url looks like cid:image0001.png)ED100.... if I were to use the GetAttachement() function I can get the image but once I download the image in the signature I need to tell the <img src="cid:image0001.pngED100.." /> in the html message body signature to be <img src="path/to/the/image" /> correct?

However, if I do that...when I reply to the message the img src is the src that I changed it to (src="path/to/my/image") now does this mean I need to attache that image again in order for the image to display properly on other email clients?

simply put, my email client is web based and I am display the message body in an iframe, I need the message to display properly with the signatures and their images.

Thanks so much for your help I really appreciate it.

**Comment by jamesiarmes** [comment]
If you are including the same signature in the body of the new message, then I believe you would need to attach the the image to the new message. Depending on your server/user settings, replying may automatically include the quoted email message, so including the original signature from the sender may not be necessary. I have never actually tried this, so I unfortunately can't verify, this is just a best guess.

**Comment by erikegon** [comment]
I have never been able to forward or reply to a message using the appropriate functions. Somehow the message is either not sent or delivers an empty page. So my workaround is to re-attach all inline and file attachments to a new message and copy the old body underneath the reply/forward text and use str_replace() to change/reformat the img-tags to reflect the format Exchange expects.

When I receive new messages I create an attachment-folder equal to md5(ItemId->Id) and copy the attachments to that location. When replying the attachments can be easily found by listing all files in that folder.

Took some time to work it all out, but it does work.

**Comment by jamesiarmes** [comment]
Have you tried the [message/reply-all](https://github.com/jamesiarmes/php-ews/blob/master/examples/message/reply-all.php) example? You shouldn't have to create a completely new message for a reply and doing so will result in Exchange and other email clients from not recognizing the message as being a reply.
